lib/dao.js

```diff
@@ -1597,6 +1598,22 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
+          self._coerce(clause, props[model].type.definition.properties)[prop];
```

Comment:
the `model-builder` attaches the `model.definiton` property to every single model. So, if the model has a type of `ModelConstructor` then we can be certain that those properties exist. See: [lib/model-builder.js:276](https://github.com/strongloop/loopback-datasource-juggler/blob/2.x/lib/model-builder.js#L276)